src/partition/growfs.c

```diff
@@ -25,13 +24,15 @@
 #include "parse-util.h"
 #include "pretty-print.h"
 #include "resize-fs.h"
+#include "sd-device.h"
 
 static const char *arg_target = NULL;
 static bool arg_dry_run = false;
 
 #if HAVE_LIBCRYPTSETUP
 static int resize_crypt_luks_device(dev_t devno, const char *fstype, dev_t main_devno) {
-        _cleanup_free_ char *devpath = NULL, *main_devpath = NULL;
+        _cleanup_(sd_device_unrefp) sd_device *main_dev = NULL, *dev = NULL;
+        _cleanup_free_ const char *devpath = NULL, *main_devpath = NULL;
```

Comment:
Please do not use `_cleanup_free_` attributes for the paths obtained by `sd_device_get_devpath()` are owned by the sd_device object.